python/lsst/meas/astrom/loadReferenceObjects.py

```diff
@@ -0,0 +1,166 @@
+        omitted unless defaultFilter is specified in the config
+    - default_fluxSigma (optional): brightness standard deviation to use if no camera filter is available
+        (magnitude); omitted unless defaultFilter is specified in the config and the corresponding
+        fluxSigma field exists
+    - photometric (optional): is the object usable for photometric calibration?
+    - resolved (optional): is the object spatially resolved?
+    - variable (optional): does the object have variable brightness?
+
+    Implementations must subclass this class, override the methods marked abstractmethod,
+    and will typically override the value of ConfigClass with a task-specific config class.
+    """
+    __metaclass__ = abc.ABCMeta
+    ConfigClass = LoadReferenceObjectsConfig
+    _DefaultName = "LoadReferenceObjects"
+
+    def getFluxField(self, cameraFilter=None, schema=None):
+        """Get flux field name
+
+        if cameraFilter is specified:
+            name = self.config.filterMap[cameraFilter] if present, else cameraFilter
+            return <name> + "_flux"
+        else:
+            return self.config.defaultFilter + "_flux", if specified, else throw RuntimeError
+
+        @param[in] cameraFilter  name of camera filter
+        @param[in] schema  if present then checks that the name exists; if None then no test is performed
+        @return flux field name
+
+        @throw RuntimeError if cameraFilter not specified and no default is present in the config
+        @throw RuntimeError if cameraFilter and schema specified and the resulting field does not exist
+        """
+        if cameraFilter:
+            refFilter = self.config.filterMap.get(cameraFilter, cameraFilter)
+        else:
+            if not self.config.defaultFilter:
+                raise RuntimeError("cameraFilter=%s and and no default filter" % (cameraFilter,))
+            refFilter = self.config.defaultFilter
+
+        fluxField = refFilter + "_flux"
+        if schema is not None and refFilter not in schema:
+            raise RuntimeError("cameraFilter=%s but flux field %s not present in schema" %
+                (cameraFilter, fluxField))
+        return fluxField
+
+    @abc.abstractmethod
+    def loadObjectsInBBox(self, bbox, wcs, filterName=None, calib=None):
+        """!Load reference objects that overlap a pixel-based rectangular region
+
+        The search algorith works by searching in a region in sky coordinates whose center is the center
+        of the bbox and radius is large enough to just include all 4 corners of the bbox.
+        Stars that lie outside the bbox are then trimmed from the list.
+
+        @param[in] bbox  bounding box for pixels (an lsst.afw.geom.Box2I or Box2D)
+        @param[in] wcs  WCS (an lsst.afw.image.Wcs)
+        @param[in] filterName  name of filter, or None or blank for the default filter
+        @param[in] calib  calibration, or None if unknown
```

Comment:
I came to the opposite conclusion: it was easier and more natural to support calib being None if the reference loader continued to return flux instead of magnitudes. We should discuss this further.